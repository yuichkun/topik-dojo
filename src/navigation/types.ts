/**
 * ナビゲーション型定義
 */

import type { NativeStackScreenProps } from '@react-navigation/native-stack';

// ルートスタックのパラメータ定義
export type RootStackParamList = {
  Top: undefined;
  LearningModeSelection: {
    level: number;
  };
  // TODO: 他の画面のパラメータ定義を追加
  // UnitSelection: { level: number };
  // TestModeSelection: { level: number };
  // Results: { level: number };
  // Learning: { level: number; unitId: string };
  // ListeningTest: { level: number; unitId: string };
  // ReadingTest: { level: number; unitId: string };
  // Review: undefined;
};

// 各画面のプロップス型定義
export type TopScreenProps = NativeStackScreenProps<RootStackParamList, 'Top'>;
export type LearningModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'LearningModeSelection'>;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}