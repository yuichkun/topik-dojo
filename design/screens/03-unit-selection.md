# ユニット選択画面 仕様書

## 画面概要
選択された級の学習用ユニットを選択する画面。各ユニットは10問で構成される。

## UI参考資料
[元画像](../../docs/original-materials/ユニット選択.png)

## レイアウト構成
- **ヘッダー**: 選択された級の表示 + 戻るボタン
- **メインエリア**: ユニット選択ボタン一覧（グリッド表示）

## UI要素詳細

### ヘッダー部分
- **級表示**: 選択された級を表示（例：「1級 学習」）
- **戻るボタン**: 学習モード選択画面へ戻る
- **配置**: 画面上部

### ユニット選択ボタン
- **表示形式**: 「1-10」「11-20」「21-30」...の形式
- **配置**: グリッド形式で複数列表示
- **スタイル**: 統一されたボタンデザイン
- **タップ動作**: 選択したユニットの学習画面へ遷移
- **遷移先**: `05-learning.md`

### ユニット数の計算
各級のユニット数（requirements.mdより）：
- **1級**: 40ユニット（1-10, 11-20, ..., 391-400）
- **2級**: 140ユニット（1-10, 11-20, ..., 1391-1400）
- **3級**: 200ユニット（1-10, 11-20, ..., 1991-2000）
- **4級**: 200ユニット（1-10, 11-20, ..., 1991-2000）
- **5級**: 300ユニット（1-10, 11-20, ..., 2991-3000）
- **6級**: 300ユニット（1-10, 11-20, ..., 2991-3000）

## 状態管理

### 級情報の保持
- **データ**: 前画面から渡される級情報
- **表示**: ヘッダー部分に級を表示
- **引き継ぎ**: 次の学習画面に級+ユニット情報を渡す

### 学習進捗の表示（将来拡張）
- **表示内容**: 各ユニットの完了状況
- **視覚表現**: 完了済み・未完了の区別
- **実装優先度**: 中（後期実装）

## ナビゲーション

### 戻るボタン
- **動作**: 学習モード選択画面へ戻る
- **遷移先**: `02-learning-mode-selection.md`

### ホームボタン
- **動作**: トップ画面への直接遷移
- **遷移先**: `01-top.md`

## スクロール対応

### 長いリスト表示
- **対応**: 縦スクロール対応（特に高級レベル）
- **パフォーマンス**: 仮想化リスト検討（ユニット数が多い場合）

## エラーハンドリング

### 級情報なし
- **対応**: デフォルトで1級を設定
- **表示**: エラーメッセージは表示しない

### データ読み込み失敗
- **対応**: 基本的なユニット表示は維持
- **フォールバック**: 静的なユニット数で表示

## 技術要件

### 必要なデータ
- 選択された級の情報
- 各級のユニット数設定
- ユーザーの学習進捗（将来拡張用）

### パフォーマンス要件
- 画面表示: 即座に表示
- スクロール: スムーズなスクロール性能

---
*ファイル作成日: 2025/6/13*
*実装担当: AIエージェント*