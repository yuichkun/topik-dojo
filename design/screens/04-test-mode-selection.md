# テストモード選択画面 仕様書

## 画面概要
選択された級に対して、リスニングテストかリーディングテストかを選択する画面。

## UI参考資料
[元画像](../../docs/original-materials/テストモード選択.png)

## レイアウト構成
- **ヘッダー**: 選択された級の表示 + 戻るボタン
- **メインエリア**: 2つのテストモード選択ボタン（リスニング/リーディング）

## UI要素詳細

### ヘッダー部分
- **級表示**: 選択された級を表示（例：「1級 テスト」）
- **戻るボタン**: 学習モード選択画面へ戻る
- **配置**: 画面上部

### リスニングテストボタン
- **テキスト**: 「リスニング」
- **配置**: 上部のボタン
- **機能**: 音声を聞いて日本語訳を4択から選択
- **タップ動作**: テストユニット選択画面へ遷移（リスニングモード）
- **遷移先**: `05-test-unit-selection.md?mode=listening`

### リーディングテストボタン
- **テキスト**: 「リーディング」
- **配置**: 下部のボタン
- **機能**: ハングル文字を見て日本語訳を4択から選択
- **タップ動作**: テストユニット選択画面へ遷移（リーディングモード）
- **遷移先**: `05-test-unit-selection.md?mode=reading`

### ボタンスタイル
- **デザイン**: 2つのボタンは同じサイズ・同じスタイル
- **区別**: アイコンや色で リスニング/リーディング を視覚的に区別
- **レスポンシブ**: タップ時の視覚的フィードバック

## 状態管理

### 級情報の保持
- **データ**: 前画面から渡される級情報
- **表示**: ヘッダー部分に級を表示
- **引き継ぎ**: 次のテストユニット選択画面に級+テストモード情報を渡す

### テストモード情報
- **リスニング**: `mode=listening`
- **リーディング**: `mode=reading`
- **用途**: 次の画面でテスト形式を決定

## ナビゲーション

### 戻るボタン
- **動作**: 学習モード選択画面へ戻る
- **遷移先**: `02-learning-mode-selection.md`

### ホームボタン
- **動作**: トップ画面への直接遷移
- **遷移先**: `01-top.md`

## エラーハンドリング

### 級情報なし
- **対応**: デフォルトで1級を設定
- **表示**: エラーメッセージは表示しない

### 無効なテストモード選択
- **対応**: デフォルトでリーディングモードを設定
- **ログ**: デバッグ用にエラーログを記録

## 技術要件

### 必要なデータ
- 選択された級の情報
- テストモード選択の状態管理

### パフォーマンス要件
- 画面表示: 即座に表示
- ボタンレスポンス: タップ後即座に遷移

---
*ファイル作成日: 2025/6/13*
*実装担当: AIエージェント*