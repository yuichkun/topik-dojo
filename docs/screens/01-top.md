# トップ画面 仕様書

## 画面概要
アプリ起動時に最初に表示される画面。ユーザーが学習する級を選択するか、復習モードを開始するかを選択できる。

## UI参考資料
[元画像](../../docs/original-materials/top.png)

## レイアウト構成
- **ヘッダー**: アプリタイトル「TOPIK道場」
- **上部エリア**: 復習ボタン
- **メインエリア**: 級選択ボタン（1級〜6級）

## UI要素詳細

### アプリタイトル
- **テキスト**: 「TOPIK道場」
- **配置**: 画面上部中央
- **スタイル**: アプリのメインタイトル

### 級選択ボタン（1級〜6級）
- **表示**: 縦に並んだ6つのボタン
- **テキスト**: 「1級」「2級」「3級」「4級」「5級」「6級」
- **配置**: 画面中央部に縦配置
- **スタイル**: 統一されたボタンデザイン
- **タップ動作**: 選択した級の学習モード選択画面へ遷移
- **遷移先**: `02-learning-mode-selection.md`

### 復習ボタン
- **テキスト**: 「復習 (X語)」（Xは復習対象の単語数）
- **配置**: アプリタイトル直下（級選択ボタンより上部）
- **スタイル**: 級選択ボタンと区別できるデザイン（色・形状等で差別化）
- **重要度**: 級選択より優先表示（復習の方が重要）
- **表示例**: 
  - 復習対象あり: 「復習 (15語)」
  - 復習対象なし: 「復習 (0語)」（グレーアウト）
- **タップ動作**: 復習画面へ直接遷移
- **遷移先**: `08-review.md`

## 状態管理

### 復習対象数表示
- **表示場所**: 
  - アプリアイコンバッジ: 復習が必要な単語数
  - 復習ボタン内: 「復習 (X語)」形式で表示
- **表示内容**: 復習が必要な単語数
- **更新タイミング**: アプリ起動時

### データ取得処理
- 起動時にローカルデータベースから復習対象数を計算
- SRSアルゴリズムに基づいて本日復習すべき単語数を算出

## エラーハンドリング

### データ読み込み失敗
- **対応**: デフォルト画面表示（級選択は可能）
- **表示**: エラーメッセージは表示しない（ユーザビリティ重視）

### 復習データなし
- **対応**: 復習ボタンをグレーアウト（無効化）
- **条件**: 復習対象が0件の場合
- **動作**: タップ不可、視覚的に無効状態を表示

## 技術要件

### 必要なデータ
- ローカルデータベースへのアクセス
- SRS計算用のユーザー進捗データ

### パフォーマンス要件
- 起動時間: 2秒以内でメイン画面表示
- バッジ更新: アプリ起動と同時に実行

---
*ファイル作成日: 2025/6/13*
*実装担当: AIエージェント*