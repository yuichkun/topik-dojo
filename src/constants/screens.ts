/**
 * 画面名の定数定義
 */

export const SCREEN_NAMES = {
  TOP: 'Top',
  LEARNING_MODE_SELECTION: 'LearningModeSelection',
  UNIT_SELECTION: 'UnitSelection',
  LEARNING: 'Learning',
  TEST_MODE_SELECTION: 'TestModeSelection',
  TEST_UNIT_SELECTION: 'TestUnitSelection',
  READING_TEST: 'ReadingTest',
  LISTENING_TEST: 'ListeningTest',
  REVIEW: 'Review',
  // TODO: 他の画面名を追加
  // RESULTS: 'Results',
} as const;