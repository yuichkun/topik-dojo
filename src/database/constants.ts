// テーブル名の定数定義（型安全性向上）
export enum TableName {
  UNITS = 'units',
  WORDS = 'words',
  SRS_MANAGEMENT = 'srs_management',
  LEARNING_STATUS = 'learning_status',
  TEST_RESULTS = 'test_results',
  TEST_QUESTIONS = 'test_questions',
  REVIEW_HISTORY = 'review_history',
  LEARNING_PROGRESS = 'learning_progress',
}

// データベース設定の定数定義
export const DATABASE_CONFIG = {
  name: 'TopikDojo',
  testName: ':memory:',
} as const;