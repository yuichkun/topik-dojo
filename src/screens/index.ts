/**
 * 画面コンポーネントのエクスポート
 */

export { default as TopScreen } from './TopScreen';
export { default as LearningModeSelectionScreen } from './LearningModeSelectionScreen';
export { default as UnitSelectionScreen } from './UnitSelectionScreen';
export { default as LearningScreen } from './LearningScreen';
export { default as TestModeSelectionScreen } from './TestModeSelectionScreen';
export { default as TestUnitSelectionScreen } from './TestUnitSelectionScreen';
export { default as ReadingTestScreen } from './ReadingTestScreen';
export { default as ListeningTestScreen } from './ListeningTestScreen';
export { default as ReviewScreen } from './ReviewScreen';