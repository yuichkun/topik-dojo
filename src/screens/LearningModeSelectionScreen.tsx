/**
 * 学習モード選択画面
 * 選択された級に対して、学習・テスト・成績確認のいずれかのモードを選択する画面
 */

import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  SafeAreaView,
  StatusBar,
  useColorScheme,
  Alert,
} from 'react-native';
import { LearningModeSelectionScreenProps } from '../navigation/types';
import { SCREEN_NAMES } from '../constants/screens';

const LearningModeSelectionScreen: React.FC<LearningModeSelectionScreenProps> = ({
  route,
  navigation,
}) => {
  const { level } = route.params;
  const isDarkMode = useColorScheme() === 'dark';

  // 戻るボタンのタップハンドラ
  const handleBackPress = () => {
    navigation.goBack();
  };

  // 学習ボタンのタップハンドラ
  const handleLearningPress = () => {
    navigation.navigate(SCREEN_NAMES.UNIT_SELECTION, { level });
  };

  // テストボタンのタップハンドラ
  const handleTestPress = () => {
    // TODO: テストモード選択画面への遷移 (04-test-mode-selection)
    Alert.alert('テストモード', `${level}級のテストモードを選択してください`);
  };

  // 成績ボタンのタップハンドラ
  const handleResultsPress = () => {
    // TODO: 成績確認画面への遷移 (10-results)
    Alert.alert('成績確認', `${level}級の成績を確認します`);
  };

  return (
    <SafeAreaView className="flex-1 bg-white">
      <StatusBar 
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor="#ffffff"
      />
      
      {/* ヘッダー */}
      <View className="flex-row items-center justify-between px-4 py-4 border-b border-gray-200">
        <TouchableOpacity
          className="px-3 py-2"
          onPress={handleBackPress}
        >
          <Text className="text-lg text-gray-600">← 戻る</Text>
        </TouchableOpacity>
        
        <Text className="text-xl font-bold text-gray-800">{level}級</Text>
        
        <View className="w-16" />
      </View>

      {/* メインコンテンツ */}
      <View className="flex-1 justify-center px-8">
        
        {/* 学習ボタン */}
        <TouchableOpacity
          className="bg-blue-50 border-2 border-blue-500 rounded-lg py-6 mb-6 items-center"
          onPress={handleLearningPress}
        >
          <Text className="text-xl font-bold text-blue-600">学習</Text>
          <Text className="text-sm text-blue-500 mt-1">単語カード形式での語彙学習</Text>
        </TouchableOpacity>

        {/* テストボタン */}
        <TouchableOpacity
          className="bg-orange-50 border-2 border-orange-500 rounded-lg py-6 mb-6 items-center"
          onPress={handleTestPress}
        >
          <Text className="text-xl font-bold text-orange-600">テスト</Text>
          <Text className="text-sm text-orange-500 mt-1">リスニング・リーディングテスト</Text>
        </TouchableOpacity>

        {/* 成績ボタン */}
        <TouchableOpacity
          className="bg-green-50 border-2 border-green-500 rounded-lg py-6 mb-6 items-center"
          onPress={handleResultsPress}
        >
          <Text className="text-xl font-bold text-green-600">成績</Text>
          <Text className="text-sm text-green-500 mt-1">テスト結果の確認・グラフ表示</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default LearningModeSelectionScreen;