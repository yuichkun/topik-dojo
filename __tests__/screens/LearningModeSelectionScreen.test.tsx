/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Alert, TouchableOpacity, Text, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LearningModeSelectionScreen from '../../src/screens/LearningModeSelectionScreen';
import { RootStackParamList } from '../../src/navigation/types';
import { SCREEN_NAMES } from '../../src/constants/screens';

// AlertのスパイFunction
jest.spyOn(Alert, 'alert');

const Stack = createNativeStackNavigator<RootStackParamList>();

// Mock screens for navigation targets
const MockUnitSelectionScreen = () => <View><Text>Mock Unit Selection</Text></View>;
const MockTestModeSelectionScreen = () => <View><Text>Mock Test Mode Selection</Text></View>;
const MockResultsScreen = () => <View><Text>Mock Results</Text></View>;

// テスト用のナビゲーションコンテナーを作成
const createTestNavigationContainer = (level: number) => {
  const TestNavigationContainer = () => (
    <NavigationContainer>
      <Stack.Navigator initialRouteName={SCREEN_NAMES.LEARNING_MODE_SELECTION}>
        <Stack.Screen
          name={SCREEN_NAMES.LEARNING_MODE_SELECTION}
          component={LearningModeSelectionScreen}
          initialParams={{ level }}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={SCREEN_NAMES.UNIT_SELECTION}
          component={MockUnitSelectionScreen}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={SCREEN_NAMES.TEST_MODE_SELECTION}
          component={MockTestModeSelectionScreen}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={SCREEN_NAMES.RESULTS}
          component={MockResultsScreen}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
  return TestNavigationContainer;
};

describe('LearningModeSelectionScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly with level 1', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!).toBeDefined();
  });

  test('displays selected level in header', async () => {
    const TestContainer = createTestNavigationContainer(3);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // Text要素の内容を確認 - 配列形式で [3, '級'] となっている
    const textElements = component!.root.findAllByType(Text);
    const levelText = textElements.find(element => {
      const children = element.props.children;
      return (
        Array.isArray(children) && children[0] === 3 && children[1] === '級'
      );
    });
    expect(levelText).toBeDefined();
  });

  test('displays all three mode buttons', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 3つのモードボタンを確認
    const learningButton = component!.root.findByProps({ children: '学習' });
    const testButton = component!.root.findByProps({ children: 'テスト' });
    const resultsButton = component!.root.findByProps({ children: '成績' });

    expect(learningButton).toBeDefined();
    expect(testButton).toBeDefined();
    expect(resultsButton).toBeDefined();
  });

  test('learning button can be pressed', async () => {
    const TestContainer = createTestNavigationContainer(2);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 学習ボタンを見つけてタップ
    const touchableOpacities = component!.root.findAllByType(TouchableOpacity);
    const learningButton = touchableOpacities.find(button => {
      try {
        const text = button.findByProps({ children: '学習' });
        return text !== undefined;
      } catch {
        return false;
      }
    });

    expect(learningButton).toBeDefined();

    // ボタンが押せることを確認（エラーが発生しないことを確認）
    await ReactTestRenderer.act(async () => {
      learningButton!.props.onPress();
    });

    // 学習ボタンは現在ナビゲーションするため、Alertは呼ばれない
    expect(Alert.alert).not.toHaveBeenCalled();
  });

  test('test button navigates to test mode selection when pressed', async () => {
    const TestContainer = createTestNavigationContainer(4);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // テストボタンを見つけてタップ
    const touchableOpacities = component!.root.findAllByType(TouchableOpacity);
    const testButton = touchableOpacities.find(button => {
      try {
        const text = button.findByProps({ children: 'テスト' });
        return text !== undefined;
      } catch {
        return false;
      }
    });

    expect(testButton).toBeDefined();

    // ナビゲーションはテスト環境では実際には動作しないが、エラーが出ないことを確認
    await ReactTestRenderer.act(async () => {
      testButton!.props.onPress();
    });
  });

  test('results button navigates to results screen when pressed', async () => {
    const TestContainer = createTestNavigationContainer(5);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 成績ボタンを見つけてタップ
    const touchableOpacities = component!.root.findAllByType(TouchableOpacity);
    const resultsButton = touchableOpacities.find(button => {
      try {
        const text = button.findByProps({ children: '成績' });
        return text !== undefined;
      } catch {
        return false;
      }
    });

    expect(resultsButton).toBeDefined();

    await ReactTestRenderer.act(async () => {
      resultsButton!.props.onPress();
    });

    // Should navigate to results screen instead of showing alert
    // Verify that the MockResultsScreen is rendered
    await ReactTestRenderer.act(async () => {
      const mockResultsText = component!.root.findByProps({ children: 'Mock Results' });
      expect(mockResultsText).toBeDefined();
    });
  });

  test('back button is present', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    const backButton = component!.root.findByProps({ children: '← 戻る' });
    expect(backButton).toBeDefined();
  });

  test('matches snapshot', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });
});
