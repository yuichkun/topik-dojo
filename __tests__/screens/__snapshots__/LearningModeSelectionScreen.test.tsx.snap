// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningModeSelectionScreen matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      activityState={2}
      aria-hidden={false}
      collapsable={false}
      fullScreenSwipeShadowEnabled={true}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onHeaderHeightChange={[Function]}
      onNativeDismissCancelled={[Function]}
      onSheetDetentChanged={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents={
        [
          1,
        ]
      }
      sheetCornerRadius={-1}
      sheetElevation={24}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetInitialDetent={0}
      sheetInitialDetentIndex={0}
      sheetLargestUndimmedDetent={-1}
      sheetLargestUndimmedDetentIndex={-1}
      stackPresentation="push"
      style={
        [
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ],
          {
            "zIndex": undefined,
          },
        ]
      }
      swipeDirection="horizontal"
    >
      <RNSScreenContentWrapper
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ],
          ]
        }
      >
        <RCTSafeAreaView
          className="flex-1 bg-white"
        >
          <View
            className="flex-row items-center justify-between px-4 py-4 border-b border-gray-200"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-lg text-gray-600"
              >
                ← 戻る
              </Text>
            </View>
            <Text
              className="text-xl font-bold text-gray-800"
            >
              1
              級
            </Text>
            <View
              className="w-16"
            />
          </View>
          <View
            className="flex-1 justify-center px-8"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-xl font-bold text-blue-600"
              >
                学習
              </Text>
              <Text
                className="text-sm text-blue-500 mt-1"
              >
                単語カード形式での語彙学習
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-xl font-bold text-orange-600"
              >
                テスト
              </Text>
              <Text
                className="text-sm text-orange-500 mt-1"
              >
                リスニング・リーディングテスト
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-xl font-bold text-green-600"
              >
                成績
              </Text>
              <Text
                className="text-sm text-green-500 mt-1"
              >
                テスト結果の確認・グラフ表示
              </Text>
            </View>
          </View>
        </RCTSafeAreaView>
      </RNSScreenContentWrapper>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleFontFamily="System"
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleFontFamily="System"
        largeTitleFontWeight="700"
        largeTitleHideShadow={false}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
        title="LearningModeSelection"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="System"
        titleFontWeight="600"
        topInsetEnabled={true}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
